app/main.py

```diff
@@ -1 +1,36 @@
-# write your code here
+from __future__ import annotations
+
+
+class Animal:
+
+    alive: list[Animal] = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name: str = name
+        self.health: int = max(min(health, 100), 0)
+        self.hidden: bool = hidden
+        if self.health > 0:
+            Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    def bite(self, animal: Herbivore) -> None:
+        if isinstance(animal, Herbivore) and not animal.hidden:
+            animal.health = max(animal.health - 50, 0)
```

Comment:
again it's hard to understand the code at first glance, just subtract 50 and check if health <=0